src/cloudflare/internal/images-api.ts

```diff
@@ -303,6 +327,165 @@ function chainStreams<T>(streams: ReadableStream<T>[]): ReadableStream<T> {
+          return;
+        }
+      }
+
+      buffer = offset < currentData.length ? currentData.slice(offset) : null;
+    },
+
+    flush(controller): void {
+      if (buffer && buffer.length > 0) {
+        try {
+          controller.enqueue(toBase64(buffer));
+        } catch (error) {
+          const errMsg = error instanceof Error ? error.message : String(error);
+          controller.error(new Error(`base64 encoding error: ${errMsg}`));
+        }
+      }
+      buffer = null;
+    },
+  });
+}
+
+function createBase64DecoderTransformStream(
+  maxChunkSize: number = 32 * 1024
+): TransformStream<Uint8Array, Uint8Array> {
+  if (maxChunkSize % 4 !== 0 || maxChunkSize <= 0) {
+    throw new Error('maxChunkSize must be a positive multiple of 4.');
+  }
+
+  let base64Buffer: Uint8Array | null = null;
+  const asciiDecoder = new TextDecoder('ascii');
+
+  const decodeAndEnqueueSegment = (
+    base64SegmentBytes: Uint8Array,
+    controller: TransformStreamDefaultController<Uint8Array>
+  ): boolean => {
+    try {
+      const base64SegmentString = asciiDecoder.decode(base64SegmentBytes);
+
+      const binaryString = atob(base64SegmentString);
+
+      if (binaryString.length > 0) {
+        const decodedBytes = Uint8Array.from(binaryString, (c) =>
+          c.charCodeAt(0)
+        );
+
+        controller.enqueue(decodedBytes);
+      }
+      return true;
+    } catch (error) {
+      controller.error(new Base64Error(error));
+      return false;
+    }
+  };
+
+  return new TransformStream<Uint8Array, Uint8Array>({
+    transform(chunk, controller): void {
+      base64Buffer = base64Buffer
+        ? concatUint8Arrays(base64Buffer, chunk)
+        : chunk;
+
+      while (base64Buffer && base64Buffer.length >= maxChunkSize) {
+        const base64SegmentBytes = base64Buffer.slice(0, maxChunkSize);
+        base64Buffer =
+          base64Buffer.length > maxChunkSize
+            ? base64Buffer.slice(maxChunkSize)
+            : null;
+
+        if (!decodeAndEnqueueSegment(base64SegmentBytes, controller)) {
+          base64Buffer = null;
+          return;
+        }
+      }
+
+      const remainingProcessableLength =
+        Math.floor(base64Buffer?.length ?? 0 / 4) * 4;
+      if (remainingProcessableLength > 0) {
+        if (base64Buffer) {
+          const base64SegmentBytes = base64Buffer.slice(
+            0,
+            remainingProcessableLength
+          );
+          base64Buffer =
+            base64Buffer.length > remainingProcessableLength
+              ? base64Buffer.slice(remainingProcessableLength)
+              : null;
+
+          if (!decodeAndEnqueueSegment(base64SegmentBytes, controller)) {
+            base64Buffer = null;
+            return;
+          }
+        }
+      }
+    },
+
+    flush(controller): void {
+      if (base64Buffer && base64Buffer.length > 0) {
+        if (!decodeAndEnqueueSegment(base64Buffer, controller)) {
+          // Error already handled within the helper
+        }
```

Comment:
Why do you need this?

Reply:
Just keeps symmetry with the block above - removed